timeline-chart/src/time-graph-viewport-container.ts

```diff
@@ -0,0 +1,205 @@
+        if (layer instanceof TimeGraphViewportLayer) {
+            layer.bindViewport(this);
+            super.addLayer(layer);
+        } else {
+            throw "You can only add TimeGraphViewportLayers to Viewports.";
+        }
+    }
+
+    updateCanvas = (newWidth: number, newHeight: number, newColor?: number, lineColor?: number) => {
+        super.updateCanvas(newWidth, newHeight, newColor, lineColor);
+        // This is where we change how the window resizing works.
+        // We can not call super and rethink the logic.
+        this.updateBackground();
+    };
+    
+    /**
+     *  Background is continuously updating with the viewRange.
+     *  Ugly but efficient.  Maybe change at some point.
+     */
+    updateBackground = () => {
+        const width = Number(this.unitController.viewRangeLength) * this.stateController.zoomFactor;
+        this.background.update({
+            position: {
+                x: -this.horizontalOffset,
+                y: 0
+            },
+            height: this.renderer.height,
+            width
+        });
+    };
+    
+    /**
+     * Horizontal Offset
+     *
+     * Setting horizontalOffset directly modifies this.stage.position.x in the setter function.
+     * 
+     * The axis is flipped and counter-intuitive.  -500 moves stage 500px to the right.
+     */
+    get horizontalOffset(): number {
+        return this._horizontalOffset;
+    }
+    
+    set horizontalOffset(x: number) {
+        this.stage.position.x = this._horizontalOffset = x;
+    }
+    
+    calculateHorizontalOffset = () => {
+        const { unitController, stateController } = this;
+        const { viewRange, worldRange } = unitController;
+        let timeOffset = Number(viewRange.start - worldRange.start);
+        this.horizontalOffset = -1 * (timeOffset * stateController.zoomFactor);
+    };
+
+    handleOnWorldRender = () => {
+        this._worldRenderedHandlers.forEach(handler => handler(this.unitController.worldRange));
+    }
+    
+    /**
+     * Performs actions that should be handled after the world is rendered.
+     * Some actions can be performed before we render the data, but some need to be performed post-render.
+     */
+    onWorldRender = (handler: (worldRange: TimelineChart.TimeGraphRange) => void) => {
+        this._worldRenderedHandlers.push(handler);
+    }
+
+    removeWorldRenderHandler = (handler: (worldRange: TimelineChart.TimeGraphRange) => void) => {
+        const index = this._worldRenderedHandlers.indexOf(handler);
+        if (index > -1) {
+            this._worldRenderedHandlers.splice(index, 1);
+        }
+    }
+
+    destroy = () => {
+        this.unitController.removeViewRangeChangedHandler(this._onViewRangeChanged);
+        this.removeWorldRenderHandler(this._onWorldRender);  // Why not.
+        super.destroy();
+    }
+
+
+    // Debugging functions
+    // These can be useful for future feature implementations. 
+    debug = (): any => {
```

Comment:
They can be removed.  I'd like to keep them because they can be helpful for future work, but ultimately this is up to you.  I can also save this code somewhere else and put it back in if needed for future work.

Reply:
These have been removed.